test/sql/include/plan_expand_hypertable_chunks_in_query.sql

```diff
@@ -0,0 +1,19 @@
+-- Copyright (c) 2016-2018  Timescale, Inc. All Rights Reserved.
+--
+-- This file is licensed under the Apache License,
+-- see LICENSE-APACHE at the top level directory.
+
+--we want to see how our logic excludes chunks
+--and not how much work constraint_exclusion does
+SET constraint_exclusion = 'off';
+
+-- explicit chunk exclusion
+:PREFIX SELECT * FROM hyper WHERE chunks_in(hyper, ARRAY[1,2]) ORDER BY value;
+:PREFIX SELECT * FROM (SELECT * FROM hyper h WHERE chunks_in(h, ARRAY[1,2,3])) T ORDER BY value;
+:PREFIX SELECT * FROM hyper WHERE chunks_in(hyper, ARRAY[1,2,3]) AND time < 10 ORDER BY value;
+:PREFIX SELECT * FROM hyper_ts h JOIN tag on (h.tag_id = tag.id ) WHERE chunks_in(h, ARRAY[116]) AND time < to_timestamp(10) AND device_id = 'dev1' ORDER BY value;
+:PREFIX SELECT * FROM hyper_w_space h1 JOIN hyper_ts h2 ON h1.device_id=h2.device_id WHERE chunks_in(h1, ARRAY[104,105]) AND chunks_in(h2, ARRAY[116,117]) ORDER BY h1.value;
+\set ON_ERROR_STOP 0
+:PREFIX SELECT * FROM hyper WHERE chunks_in(hyper, ARRAY[1,2]) AND chunks_in(hyper, ARRAY[2,3]);
+:PREFIX SELECT * FROM hyper WHERE chunks_in(2, ARRAY[1]);
+:PREFIX SELECT * FROM hyper WHERE time < 10 OR chunks_in(hyper, ARRAY[1,2]);
```

Comment:
I'd like to see a self-join here:
SELECT * FROM hyper h1, hyper h2 WHERE chunks_in(h1, ARRAY[1,2]) AND chunks_in(h2, ARRAY[2,3]);

Reply:
I'd like to see at least 1 query here that uses index scan.

Reply:
Can you explain why is that needed in the scope of proposed changes?

Reply:
Because you pass the full row into the function decisions about row inclusions can't be made from index alone, so this implementation is going to prevent index-only scans, just want to make sure that it doesnt also prevent index scans. It might be best to remove the chunks_in call from the plan once we have selected the chunks.

Reply:
That's not true. This function is not preventing neither index-only or index scans. I've added two tests that prove that.

Reply:
@niksajakovljevic Sven is actually right here. Your test `SELECT time FROM hyper WHERE time=0 AND chunks_in(hyper, ARRAY[1]);` should use an INDEX ONLY but in your test it's running an INDEX scan. 

The reason is probably that its interpreting  the `chunks_in(hyper, ARRAY[1])` as a "whole row" reference to the hyper row (pretty much this is probably interpreted by the planner as a RECORD type for the entire row). Which means it would need to fetch the entire row and not just the "time" column. But the reference to the entire row means it can't just use the index (which has the time column but not the entire row).

I think the solution here is to remove the `chunks_in(hyper, ARRAY[1])` from the plan after processing it. I think that would remove the reference before the "can an index-only scan be used" check happens. But obviously the test will show if this modification succeeds.

Reply:
Yes, I see. I'll remove `chunks_in` from the query tree.